tools/benchmark/tests-code.js

```diff
@@ -0,0 +1,606 @@
+      b.test.testDeferredFunction(nonParsableProgram, {}, 'loadParsedPyret',function(result){
+        passed = result;
+        flag = true;
+      });
+    });
+
+    waitsFor(function(){
+      return flag;
+    }, 'failure',5000);
+
+    runs(function(){
+      expect(passed).toBe(false);
+    });
+  });
+});
+
+describe('setup', function(){
+  it('sets up parsed ast and loaded js module', function(){
+    var passed, flag;   
+
+    runs(function(){
+      debugger;
+      passed = false;
+      flag = false;
+      b.test.testSetup(validProgram, {}, function(result){
+        passed = result;
+        flag = true;
+      });
+    });
+
+    waitsFor(function(){
+      return flag;
+    }, 'failure',5000);
+
+    runs(function(){
+      expect(passed).toBe(true);
+    });
+  });
+});
+
+describe('evalLoadedPyret', function(){
+  it('passes on a valid program', function() {
+    var passed, flag;   
+
+    runs(function(){
+      passed = false;
+      flag = false;
+      b.test.testDeferredFunction(validProgram, {}, 'evalLoadedPyret',function(result){
+        passed = result;
+        flag = true;
+      });
+    });
+
+    waitsFor(function(){
+      return flag;
+    }, 'failure',5000);
+
+    runs(function(){
+      expect(passed).toBe(true);
+    });
+  });
+
+  xit('passes on a valid (long) program', function() {
+    var passed, flag;   
+
+    runs(function(){
+      passed = false;
+      flag = false;
+      b.test.testDeferredFunction(longProgram, {}, 'evalLoadedPyret',function(result){
+        passed = result;
+        flag = true;
+      });
+    });
+
+    waitsFor(function(){
+      return flag;
+    }, 'failure',5000);
+
+    runs(function(){
+      expect(passed).toBe(true);
+    });
+  });
+
+  it('fails on an invalid program', function() {
+    var passed, flag;   
+
+    runs(function(){
+      passed = false;
+      flag = false;
+      b.test.testDeferredFunction(invalidProgram, {}, 'evalLoadedPyret',function(result){
+        passed = result;
+        flag = true;
+      });
+    });
+
+    waitsFor(function(){
+      return flag;
+    }, 'failure',5000);
+
+    runs(function(){
+      expect(passed).toBe(false);
+    });
+  }); 
+
+  it('fails on a nonparsable program', function() {
+    var passed, flag;   
+
+    runs(function(){
+      passed = false;
+      flag = false;
+      b.test.testDeferredFunction(nonParsableProgram, {}, 'evalLoadedPyret',function(result){
+        passed = result;
+        flag = true;
+      });
+    });
+
+    waitsFor(function(){
+      return flag;
+    }, 'failure',5000);
+
+    runs(function(){
+      expect(passed).toBe(false);
+    });
+  });
+});
+
+describe('createSuite', function(){
+  it('creates an instance of Benchmark.Suite with the correct number of benchmarks', function(){
+    expect(b.test.testCreateSuite()).toBe(true);
+  });
+});
+
+describe('runBenchmarks', function(){
+  it('returns array of objects with correct field types', function() {
+    debugger;
```

Comment:
Yes. I've since locally removed all the uncommented debugger statements I could find, which will be reflected in a subsequent push.